src/BenchmarkDotNet/Code/ArrayParam.cs

```diff
@@ -7,6 +7,12 @@
 
 namespace BenchmarkDotNet.Code
 {
+    internal static class ArrayParam
```

Comment:
There is no need to introduce a separate class; it will be better to just add a new private static method in `ArrayParam<T>`.

Reply:
In that case a generic parameter must be specified when the method is called from `SmartArgument` for example. Of course it can be `object`, but using a shared instance when it isn't needed is a bad idea, you know.

Reply:
Oh, got it. Ok, let's keep the current implementation. I will refactor it later (will try to find a better util class).